src/protocol/attribution/aggregate_credit.rs

```diff
@@ -274,15 +278,14 @@ async fn sort_by_aggregation_bit<F: Field>(
 ) -> Result<Vec<CappedCreditsWithAggregationBit<F>>, Error> {
     // Since aggregation_bit is a 1-bit share of 1 or 0, we'll just extract the
     // field and wrap it in another vector.
-    let aggregation_bits = &[input
+    let aggregation_bits = vec![input
         .iter()
-        .map(|x| x.aggregation_bit.clone())
+        .map(|x| vec![x.aggregation_bit.clone()])
```

Comment:
is it possible to avoid allocating a vector here? would `&[x.aggregation_bit]` serve the same purpose?

Reply:
This needs another bunch of refactoring since this is an input to sort which is called from multiple places. Lets revisit this in the next PR?